OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -1003,6 +1003,64 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					}
 				}
 
+				if (engineVersion < 20171006)
+				{
+					// CanPowerDown now provides a condition instead of triggering Actor.Disabled
+					var canPowerDown = node.Value.Nodes.FirstOrDefault(n => n.Key.StartsWith("CanPowerDown", StringComparison.Ordinal));
+					if (canPowerDown != null)
+					{
+						canPowerDown.Value.Nodes.Add(new MiniYamlNode("PowerdownCondition", "powerdown"));
+
+						var image = canPowerDown.Value.Nodes.FirstOrDefault(n => n.Key == "IndicatorImage");
+						var seq = canPowerDown.Value.Nodes.FirstOrDefault(n => n.Key == "IndicatorSequence");
+						var pal = canPowerDown.Value.Nodes.FirstOrDefault(n => n.Key == "IndicatorPalette");
+						var imageValue = image != null ? FieldLoader.GetValue<string>("IndicatorImage", image.Value.Value) : "poweroff";
+						var seqValue = seq != null ? FieldLoader.GetValue<string>("IndicatorSequence", seq.Value.Value) : "offline";
+						var palValue = pal != null ? FieldLoader.GetValue<string>("IndicatorPalette", pal.Value.Value) : "chrome";
+
+						var indicator = new MiniYamlNode("WithDecoration@POWERDOWN", "");
+						indicator.Value.Nodes.Add(new MiniYamlNode("Image", imageValue));
+						indicator.Value.Nodes.Add(new MiniYamlNode("Sequence", seqValue));
+						indicator.Value.Nodes.Add(new MiniYamlNode("Palette", palValue));
+						indicator.Value.Nodes.Add(new MiniYamlNode("RequiresCondition", "powerdown"));
+						indicator.Value.Nodes.Add(new MiniYamlNode("ReferencePoint", "Center"));
+
+						node.Value.Nodes.Add(indicator);
+						if (image != null)
+							canPowerDown.Value.Nodes.Remove(image);
+						if (seq != null)
+							canPowerDown.Value.Nodes.Remove(seq);
+						if (pal != null)
+							canPowerDown.Value.Nodes.Remove(pal);
+
+						Console.WriteLine("CanPowerDown now provides a condition instead of disabling the actor directly.");
+						Console.WriteLine("Review your condition setup to make sure all relevant traits are disabled by that condition.");
+						Console.WriteLine("Look at the official mods if you need examples.");
+					}
+
+					// RequiresPower has been replaced with GrantConditionOnPowerState.
+					var requiresPower = node.Value.Nodes.FirstOrDefault(n => n.Key == "RequiresPower");
+					if (requiresPower != null)
+					{
+						requiresPower.Key = "GrantConditionOnPowerState@LOWPOWER";
+						requiresPower.Value.Nodes.Add(new MiniYamlNode("Condition", "lowpower"));
+
+						Console.WriteLine("RequiresPower has been replaced with GrantConditionOnPowerState.");
+						Console.WriteLine("This new trait provides a condition if the current player power is valid.");
```

Comment:
"... is valid" doesn't make sense to me here.

It toggles a condition based on power state. Saying anything other than that is confusing IMO.